src/interpreter/lexical-environment/get-dot-path-from-node.ts

```diff
@@ -21,8 +21,8 @@ export function getDotPathFromNode<T extends TS.Node>(options: EvaluatorOptions<
 		return SUPER_SYMBOL;
 	} else if (typescript.isParenthesizedExpression(node)) {
 		return getDotPathFromNode({...options, node: node.expression});
-	} else if (typescript.isTypeAssertionExpression?.(node) || (!("isTypeAssertionExpression" in typescript) && (typescript as typeof TS).isTypeAssertion(node))) {
-		return getDotPathFromNode({...options, node: node.expression});
+	} else if (typescript.isTypeAssertionExpression?.(node) || (!("isTypeAssertionExpression" in typescript) && (typescript as any).isTypeAssertion(node))) {
```

Comment:
If instead of the `any` cast here, we write it like this:

```ts
// ...
else if (
		rest.typescript.isTypeAssertionExpression?.(node) ||
		(!("isTypeAssertionExpression" in rest.typescript) && (rest.typescript as {isTypeAssertion: typeof TS.isTypeAssertionExpression}).isTypeAssertion(node))
	)
```